bin/udesk

```diff
@@ -0,0 +1,96 @@
+    REMOVED_PACKAGES=( $(diff --changed-group-format='%>' \
+        --unchanged-group-format='' \
+        - ${HOME}/.config/toolbox/packages-${HOSTNAME} <<< `$LIST_ALL_PACKAGES`) )
+    ADDED_PACKAGES=( $(diff --changed-group-format='%<' \
+        --unchanged-group-format='' \
+        - ${HOME}/.config/toolbox/packages-${HOSTNAME} <<< `$LIST_ALL_PACKAGES`) )
+else
+    [ ! -d ${HOME}/.config/toolbox ] && mkdir -p ${HOME}/.config/toolbox
+    ADDED_PACKAGES=( $($LIST_ALL_PACKAGES) )
+fi
+$LIST_ALL_PACKAGES > ${HOME}/.config/toolbox/packages-${HOSTNAME}
+
+
+
+for ADDED_PACKAGE in "${ADDED_PACKAGES[@]}"
+do
+    echo "ADDED PACKAGE: ${ADDED_PACKAGE}"
+
+    #Will hold all icons we need
+    declare -a ICONS=()
+
+    PACKAGE_FILES=( $($LIST_PACKAGE_CONTENTS ${ADDED_PACKAGE}) )
+
+    for PACKAGE_FILE in "${PACKAGE_FILES[@]}"
+    do
+        if [[ "$PACKAGE_FILE" =~ .*applications.*\.desktop$ ]]
+        then
+            #Create desktop file for container host
+            APPLICATION_NAME=`basename ${PACKAGE_FILE}`
+            APPLICATION_NAME=`sed "s/.desktop$//" <<< ${APPLICATION_NAME}`
+            APPLICATION_NAME="${APPLICATION_NAME}-on-${HOSTNAME}.desktop"
+
+            sed "s/Exec=/Exec=toolbox -u -c ${HOSTNAME} -- /g" $PACKAGE_FILE | \
+                sed '/^TryExec/d' | \
+                sed "s/^Name=.*/& on ${HOSTNAME}/g" | \
+                sed "s/^Icon=/Icon=${HOSTNAME}-${ADDED_PACKAGE}-/g" \
+                > ${HOME}/.local/share/applications/${APPLICATION_NAME}
+
+            echo "#${ADDED_PACKAGE}-${HOSTNAME}" >> ${HOME}/.local/share/applications/${APPLICATION_NAME}
+            echo "ADDED DESKTOP FILE: ${HOME}/.local/share/applications/${APPLICATION_NAME}"
+
+            #Find icon to copy to container HOSTNAME
+            ICONS+=( $(sed -n "/Icon=/s/Icon=//p" $PACKAGE_FILE | \
+            xargs -I % find /usr/share/icons /usr/share/pixmaps -name "%*") )
+        fi
+    done
+
+    for ICON in "${ICONS[@]}"
+    do
+        echo $ICON
+        ICON_NAME=`basename ${ICON}`
+        ICON_PATH=`dirname ${ICON}`
+        LOCAL_ICON_PATH=`dirname ${ICON} | sed "s|^.*icons|${HOME}/.local/share/icons|" | \
+            sed "s|^.*pixmaps|${HOME}/.local/share/pixmaps|"`
+        LOCAL_ICON_NAME=${HOSTNAME}-${ADDED_PACKAGE}-${ICON_NAME}
+        [ ! -d $LOCAL_ICON_PATH ] && mkdir -p $LOCAL_ICON_PATH
+        cp ${ICON_PATH}/${ICON_NAME} ${LOCAL_ICON_PATH}/${LOCAL_ICON_NAME}
+    done
+done
+
+for REMOVED_PACKAGE in "${REMOVED_PACKAGES[@]}"
+do
+    echo "REMOVED PACKAGE: ${REMOVED_PACKAGE}"
+    APPLICATIONS=( $(grep -Ilr ${REMOVED_PACKAGE}-${HOSTNAME} ${HOME}/.local/share/applications/) )
+    for APPLICATION in "${APPLICATIONS[@]}"
+    do
+        echo "REMOVED DESKTOP FILE: ${APPLICATION}"
+        rm $APPLICATION
+    done
+    find ${HOME}/.local/share/icons -name "*${HOSTNAME}-${REMOVED_PACKAGE}-*" -delete
+    find ${HOME}/.local/share/pixmaps -name "*${HOSTNAME}-${REMOVED_PACKAGE}-*" -delete
+done
+
+exit 0
```

Comment:
Sounds like this file should be part of the toolbox container? This repository contains the infrastructure to run a toolbox container, but does not build nor include a toolbox container itself.
How should this script end up in the container?
And please add documentation, how the user has to call this script and what it makes.

Reply:
Correct. This should be run from within the container. I'll update documentation.